tap_notion/client.py

```diff
@@ -78,8 +82,8 @@ def post(self, endpoint: str, params: Dict, headers: Dict, body: Dict, path: str
         """Calls the make_request method with a prefixed method type `POST`"""
 
         headers, params = self.authenticate(headers, params)
-        self.__make_request("POST", endpoint, headers=headers, params=params, data=body, timeout=self.request_timeout)
-
+        return self.__make_request("POST", endpoint, headers=headers, params=params, json=body,
+                                   timeout=self.request_timeout)
```

Comment:
Better refactoring could have been like below,
```suggestion  
        return self.__make_request("POST",
                                   endpoint,
                                   headers=headers,
                                   params=params,
                                   json=body,
                                   timeout=self.request_timeout)
```